### DISCUSSION TITLE
Issue getting an activity to start in asp.net api and blazor server app

### DISCUSSION THREAD

**Discussion by jacob7395** [open-telemetry/opentelemetry-dotnet#5010] [Q&A]
Hi,

I'm struggling to get an activity to start, I have got a listener and my traces work fine when using the AddAspNetCoreInstrumentation extension, but If I remove that extension and try to start an activity I just get null.

My packages are:

```
  <PackageReference Include="OpenTelemetry" Version="1.6.0" />
  <PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.6.0" />
  <PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.6.0" />
```

I configure my service with:

```c#
builder.Services.AddOpenTelemetry()
    .ConfigureResource(config =>
    {
        config.AddService(serviceName ?? "not-set")
            .AddAttributes(new List<KeyValuePair<string, object>>()
            {
                new("env", enviroment)
            });
    })
    .WithTracing(tracerProviderBuilder =>
    {
        tracerProviderBuilder
            .AddSource("app")
            // .AddAspNetCoreInstrumentation(options =>
            // {
            //     options.Filter = context =>
            //     {
            //         if (filterInboundContext?.Invoke(context) ?? false) return false;
            //         
            //         if (context.Request.Path == "/") return false;
            //         
            //         return true;
            //     };
            // })
            // .AddHttpClientInstrumentation(options =>
            //     {
            //         options.FilterHttpRequestMessage = message =>
            //         {
            //             if (message.RequestUri?.ToString().Contains("grafana.net/loki") ?? false) 
            //                 return false;
            //             
            //             return true;
            //         };
            //     })
            .AddOtlpExporter(otp =>
        {
            otp.Endpoint = new Uri($"http://{ip}:4317");
        });
    });
```

I have a single static ActivitySource:

```c#
public static class ActivitySources
{
    public static readonly ActivitySource App = new("app");
}
```

I have tried starting my activity when an enpoint is hit and when a blazor page is loaded, either way I just get null back:

```c#
_activity = ActivitySources.App.StartActivity("/portal");
```

I'm running a Grafana agent in a docker container locally, that's being sent to Grafana cloud. When I enable AddHttpClientInstrumentation and AddAspNetCoreInstrumentation, I get traces sent though.

I'm sure its something simple that I'm missing so any help is much appreciated :)

**Comment by cijothomas** [comment]
https://github.com/open-telemetry/opentelemetry-dotnet/issues/1217 Check if you are hitting this issue

**Reply by jacob7395** [reply]
By setting the sampler to 

```c#
.SetSampler(new AlwaysOnSampler())
```

in my config I was able to get traces in my backend.

I have a second issue, where I'm trying to start and new activity without a parent. Bellow is some simple MediatR middle ware hat I belive should start a new activity with a new ParantContext. In my backend I expecting to see this as a new trace with a link back to my perant.

```c#
public class MediatROpenTelematryBehaviour<TRequest, TResponse> : IPipelineBehavior<TRequest, TResponse>
    where TRequest : notnull
{
    
    
    public async Task<TResponse> Handle(TRequest request, RequestHandlerDelegate<TResponse> next, CancellationToken cancellationToken)
    {
        ActivityContext currentContext = Activity.Current?.Context ?? new ActivityContext();
        
        Activity? activity = ActivitySources.App
            .StartActivity(request.ToString(), ActivityKind.Internal, parentContext: new ActivityContext(),
                links: new List<ActivityLink>()
                {
                    new ActivityLink(currentContext)
                });
        
        activity?.SetTag("mediatr.request", request.ToString());
        activity?.SetTag("mediatr.request.type", typeof(TRequest).Name);
        activity?.SetTag("mediatr.response.type", typeof(TResponse).Name);

        try
        {
            var response = await next.Invoke();
            activity?.Dispose();
            return response;
        }
        catch (Exception)
        {
            activity?.SetStatus(ActivityStatusCode.Error);
            throw;
        }
        finally
        {
            activity?.Stop();
        }
    }
}
```

But I don't, when debugging this :

![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/9060893/40e35cc8-d39d-4445-bb24-04ad50ed3147)

In my back end the link comes though but everthing is appering under the same trace ID:

![portal](https://github.com/open-telemetry/opentelemetry-dotnet/assets/9060893/8cab031a-a5df-4b14-97b6-5c87db7dc4bb)

What this means is my entire session for my blazor app just ends up as one big span

**Comment by cijothomas** [comment]
https://github.com/open-telemetry/opentelemetry-dotnet/issues/984 Sorry, but one more issue you are hitting! Have to do the workaround to create a new root activity, in the presence of an activity Activity!

**Reply by jacob7395** [reply]
Okay, I thought it would some know issue. I'll take a look in the morning.

Thanks for your help
